eng/common/scripts/job-matrix/README.md

```diff
@@ -59,55 +99,87 @@ jobs:
       PreGenerationSteps: []
 ```
 
-### A note regarding PreGenerationSteps
+To see an example of a complete pipeline definition with a job that runs your job using matrix generator, refer to
+[`/eng/common/scripts/job-matrix/samples/matrix-test.yml`](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/job-matrix/samples/matrix-test.yml).
 
-The generation template laid out above runs as its own job. A limitation of this method is that it disallows any runtime matrix customization due to the fact that an individual job clones the targeted build SHA. The stepList `PreGenerationSteps` allows users to update matrix json however they like prior to actually invoking the matrix generation. Injected steps are run after the repository checkout, but before any matrix generation is invoked.
+### Runtime matrix generation customization
+
+The "matrix generator-enabled" job laid out above runs as its own job. A limitation of this approach is that it disallows any runtime matrix customization due to the fact that an individual job clones the targeted build SHA. That is, the matrix to generate would be determined based only on the content of the used matrix json configs.
+
+To address this limitation, we introduce the [stepList](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#parameter-data-types) `PreGenerationSteps` as well as `MatrixFilters` and `MatrixReplace`.
+
+`PreGenerationSteps` allows users to update matrix config json however they like prior to actually invoking the matrix generation. Injected steps are run after the repository checkout, but before any matrix generation is invoked.
+
+`MatrixFilters` and `MatrixReplace` allow runtime adjustment of the matrix generation process as can be seen in the source of [GenerateMatrix](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/job-matrix/job-matrix-functions.ps1#L94-L95).  
+See also [Filters](#filters) and [Replace/Modify/Append Values](#replacemodifyappend-values).
 
 ## Matrix config file syntax
 
-Matrix parameters can either be a list of strings, or a set of grouped strings (represented as a hash). The latter parameter
-type is useful for when 2 or more parameters need to be grouped together, but without generating more than one matrix permutation.
+Matrix config file is a [JSON](https://www.json.org/json-en.html) file.
```

Comment:
I don't think we need to link to the json docs, but if we do, our link analyzer might catch that we are linking to a specifically english doc url, which we disallow.

```suggestion
Matrix config file is a [JSON](https://www.json.org) file.
```

Reply:
Yeah, I see your point for not needing to link to JSON docs, but I figured we might as well be newbie-friendly and link to it on the first occurrence of "JSON", in the spirit of providing full definition the first time given abbreviation is used in text.